### PR TITLE
StatusPage widget - initial release

### PR DESCRIPTION
**Pull request by matmannion**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,68 +1,74 @@
...
+
+Builds the app for production to the `build` folder.<br />
+It correctly bundles React in production mode and optimizes the build for the best performance.
+
+The build is minified and the filenames include the hashes.<br />
+Your app is ready to be deployed!
+
+See the section about [deployment](https://facebook.github.io/create-react-app/docs/deployment) for more information.
+
+### `npm run eject`
```
Comment (lol768) [inline README.md]: Some of this seems like CRA boilerplate that can go

File: public/index.html [hunk 2]
```diff
@@ -1,43 +1,445 @@
...
+
+      <main class="id7-main-content-area" id="main">
+        <header class="id7-main-content-header">
+          <div class="id7-page-title">
+            <h1>StatusPage embed widget</h1>
+          </div>
+        </header>
+
+        <div class="id7-main-content">
+          <noscript>You need to enable JavaScript to run this app.</noscript>
```
Comment (lol768) [inline public/index.html:399]: Do you need JS for this docs page tho?

File: public/index.html [hunk 3]
```diff
@@ -1,43 +1,445 @@
...
+
+          <p><img src="https://github.com/UniversityofWarwick/statuspage-widget/workflows/Build%20and%20Deploy/badge.svg" alt="Build status"></p>
```
Comment (lol768) [inline public/index.html]: Would be nice to link through to GitHub

**Review by lol768 — APPROVED** [review]

File: public/index.html (hunk 2 above)
Comment (matmannion) [reply public/index.html:399]: There's a test mode status widget on there which won't display without it

**Review by equaliser — APPROVED** [review]

File: src/StatusPageEmbed.js [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+import './StatusPageEmbed.scss';
+
+// use native browser implementation if it supports aborting
+const fetch = ('signal' in new Request('')) ? window.fetch : fetchPolyfill;
+
+const abortController = new AbortController();
+const initialState = {
+  status: {
+    indicator: 'none',
+    description: 'All Systems Operational'
```
Comment (nanoamp) [inline src/StatusPageEmbed.js:19]: I feel like it'd be sensible to pull the 'standard' descriptors (ASO, PDS) into constants, either in this class or externally. Slightly magic as it stands.

File: src/StatusPageEmbed.js [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+  },
+  incidents: [],
+  initialised: false,
+};
+
+const testModeResponse = {
```
Comment (nanoamp) [inline src/StatusPageEmbed.js:25]: Test-case response data embedded in the class smells a little to me.

**Pull request merged**